mantidproject/mantid
# Plot workpsace on double-click (workbench)

**Description of work.**
- Double clicking a workspace now opens the spectrum selection dialog, which inlcudes a button to plot using colorfill and the buttons to create a line plot as normal.
![image](https://user-images.githubusercontent.com/46960113/60264422-0de6cc80-98db-11e9-8b1b-4512b1530e92.png)
The colorfill button is only made visible if the user has double-clicked the workspace and will not be visible if the spectrum selection dialog is accessed after a right-click.

- Add test_spectraselectiondialog

**To test:**
Load a workspace:
- With 1 spectra
- With >1 spectra

Double click, the following should happen:
- On workspace with 1 spectra: open the spectrum is plotted
- On workspace with >1 spectra: open the spectrum selector dialog
  - Inputting a range of spectrum numbers and clicking `Ok` should plot those spectra
  - Clicking the `Colorfill` button should open a colorfill plot

Now right-click the workspace with more than one spectrum and select `Plot -> Plot Spectrum`. You should see the spectrum selection dialog, but the `Colorfill` button should not be there.

**Helper code**
```python
ws2d_1spec = CreateWorkspace(DataX=[10, 20, 30],
                             DataY=[2, 3],
                             DataE=[1, 2],
                             NSpec=1,
                             Distribution=True,
                             UnitX='Wavelength',
                             VerticalAxisUnit='DeltaE',
                             VerticalAxisValues=[4])

ws2d_histo = CreateWorkspace(DataX=[10, 20, 30, 10, 20, 30, 10, 20, 30],
                             DataY=[2, 3, 4, 5, 3, 5],
                             DataE=[1, 2, 3, 4, 1, 1],
                             NSpec=3,
                             Distribution=True,
                             UnitX='Wavelength',
                             VerticalAxisUnit='DeltaE',
                             VerticalAxisValues=[4, 6, 8])
```

Fixes #25548 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
